src/segment-loader.js

```diff
@@ -1220,11 +1215,35 @@ export default class SegmentLoader extends videojs.EventTarget {
         syncPoint.time
       );
 
-      mediaIndex = mediaSourceInfo.mediaIndex;
+      segmentMediaIndex = mediaSourceInfo.mediaIndex;
       startOfSegment = mediaSourceInfo.startTime;
     }
 
-    return this.generateSegmentInfo_(playlist, mediaIndex, startOfSegment, false);
+    const segmentInfo = this.generateSegmentInfo_(playlist, segmentMediaIndex, startOfSegment, syncRequest);
+
+    if (!segmentInfo) {
+      return;
+    }
+
+    // if this is the last segment in the playlist
+    // we are not seeking and end of stream has already been called
+    // do not re-request
+    if (this.mediaSource_ && this.playlist_ && segmentInfo.mediaIndex === this.playlist_.segments.length - 1 &&
```

Comment:
Is it possible for `this.playlist_` to be falsey here?

Reply:
in the tests it happens quite often, not sure if it will happen in the wild.